### ISSUE TITLE
Feature Request: Convert `impl`s to methods

### ISSUE THREAD

**Issue by chrismit3s** [fzyzcjy/flutter_rust_bridge#67]
**Is your feature request related to a problem? Please describe.**
So I'm not even sure if this can be implemented, but in the example, there is a struct:

```
#[derive(Debug, Clone)]
pub struct Point {
    pub x: f64,
    pub y: f64,
}
```

Say we had a corresponding `impl`

```
impl Point {
    pub fn norm(&self) -> f64 {
        (self.x * self.x + self.y * self.y).sqrt()
    }
}
```

**Describe the solution you'd like**
In the binding, this `impl` could be converted to

```
class Point {
  final double x;
  final double y;

  Point({
    required this.x,
    required this.y,
  });

  Future<float> norm() {
    [[link to norm implementation]]
  }
}
```

**Describe alternatives you've considered**
I don't know how feasible this is, especially with using rust methods in ffi, but you can actually just turn them into regular functions like this:

```
pub fn point_norm(p: Point) -> f64 {
    p.norm()
}
```

**Additional context**
This is just a thought though, and I'm happy to be proven wrong :)

**Comment by chrismit3s** [comment]
Yup, will do when I find the time :) Awesome project by the way, I've been eagerly waiting for something like this to finally come around

**Comment by fzyzcjy** [comment]
Thank you!
Before this is implemented, we may use the alternative way you have suggested. It is a bit longer, but you know, C programmers always write down that long form ;) (I just mean the length of writing function names, but this package is definitely safe unlike C)

**Comment by raphaelrobert** [comment]
Bumping this so it doesn't get locked.

I think this could become even more interesting once #243 is implemented.

**Comment by chrismit3s** [comment]
Bump

**Comment by chrismit3s** [comment]
Bump

**Comment by fzyzcjy** [comment]
Being implemented in #543
